### PR TITLE
alpha to beta snapshot API

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

The alpha snapshot API is dead.  Moving to beta.

Includes hacked version of rook/ceph that supports beta snapshot api.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Cloning now uses beta snapshot API
```

### CODE REVIEW

**Comment by mhenriks** [comment]
/hold

what to get rid of generated snapshot client.  It is only used in functional tests.

File: cluster-sync/rook-ceph/pool.yaml [hunk 1]
```diff
@@ -1,15 +1,22 @@
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: replicapool
+  namespace: rook-ceph
+spec:
+  failureDomain: host
+  replicated:
+    size: 1
```
Comment (awels) [inline cluster-sync/rook-ceph/pool.yaml:9]: We want the size to be the number of nodes in the cluster, so this is the reason we had the CephBlockPool in the .sh file where we could substitute the value based on the KUBEVIRT_NUM_NODES environment variable.

File: cluster-sync/ephemeral_provider.sh [hunk 2]
```diff
@@ -51,37 +51,20 @@ function configure_hpp() {
...
-  set +e
-  retry_counter=0
-  _kubectl get VolumeSnapshotClass
-  while [[ $? -ne "0" ]] && [[ $retry_counter -lt 60 ]]; do
-    retry_counter=$((retry_counter + 1))
-    echo "Sleep 5s, waiting for VolumeSnapshotClass CRD"
-    sleep 5
-    _kubectl get VolumeSnapshotClass
+  # wait for ceph
+  until _kubectl get cephblockpools -n rook-ceph replicapool -o jsonpath='{.status.phase}' | grep Ready; do
```
Comment (awels) [inline cluster-sync/ephemeral_provider.sh:61]: Is this enough to verify the snapshot classes are available?

Comment (mhenriks) [reply cluster-sync/ephemeral_provider.sh:61]: Unlike before, we actually create the VolumeSnapshotClass ourself.  So doesn't really make sense to wait for it.  Waiting for the replicapool to be ready assures PVCs will be able to be bound right after `cluster-sync` returns.  That was not the case before.

**Comment by mhenriks** [comment]
> Looks pretty good, a couple of questions, and I am wondering how hard it would be to also update the ember_lvm lane.

I have not looked at ember yet.  Will attempt in another PR.

File: cluster-sync/ephemeral_provider.sh (hunk 2 above)
Comment (awels) [reply cluster-sync/ephemeral_provider.sh:61]: Okay sounds good

File: cluster-sync/rook-ceph/pool.yaml (hunk 1 above)
Comment (awels) [reply cluster-sync/rook-ceph/pool.yaml:9]: True, its not that important.

**Comment by awels** [comment]
Okay ember is optional lane anyway. But it will for sure fail at this point.

**Pull request merged**